core/query/src/main/java/org/eclipse/rdf4j/query/QueryResults.java

```diff
@@ -178,6 +199,40 @@ public static GraphQueryResult limitResults(GraphQueryResult queryResult, long l
 	}
 
 	/**
+	 * Parses an RDF document and returns it as a GraphQueryResult object, with parsing done on a separate
+	 * thread in the background.<br>
+	 * IMPORTANT: As this method will spawn a new thread in the background, it is vitally important that it be
```

Comment:
Editorial: change "...that it be closed..." to "...that the resulting GraphQueryResult be closed...". Also perhaps this warning should be moved (or repeated) in `@return` javadoc